lib/c-api/wasmer_wasm.h

```diff
@@ -457,6 +497,88 @@ void wasm_module_name(const wasm_module_t *module, wasm_name_t *out);
  */
 bool wasm_module_set_name(wasm_module_t *module, const wasm_name_t *name);
 
+#if defined(WASMER_WASI_ENABLED)
+/**
+ * Non-standard function to get the module name of a
+ * `wasm_named_extern_t`.
+ */
+const wasm_name_t *wasm_named_extern_module(const wasm_named_extern_t *named_extern);
```

Comment:
This may be a wider problem but we need to mark things as `own` in our extension to the Wasm C API, otherwise it's ~really hard~ impossible to know what needs to be freed by who

Reply:
Yup. I just don't know how to add attributes inside the C headers from Rust. I'm not even sure if it's doable.

Reply:
I'll explain that in the documentation.

Reply:
(in this case, ownership isn't transferred by the way)

Reply:
While documenting onwership, I found a memory leak. We said in `wasi_get_imports` and in `wasi_get_unordered_imports` that the ownership of the `wasi_env_t` parameter is taken by the function, which wasn't the case. It was memory leak.

It's not fixed.